PyRDF/backend/Local.py

```diff
@@ -52,22 +52,21 @@ def execute(self, generator):
 
         # if the RDataFrame has not been created yet or if a new one
         # is created by the user in the same session
-        if (not self.pyroot_rdf) or \
-           (self.pyroot_rdf is not generator.head_node):
+        if not self.pyroot_rdf or self.pyroot_rdf is not generator.head_node:
             self.pyroot_rdf = ROOT.ROOT.RDataFrame(*generator.head_node.args)
 
         values = mapper(self.pyroot_rdf)  # Execute the mapper function
 
         # Get the action nodes in the same order as values
         nodes = generator.get_action_nodes()
 
-        values[0].GetValue()  # Trigger event-loop
-
-        for i in range(len(values)):
+        for node, value in zip(nodes, values):
             # Set the obtained values and
             # 'RResultPtr's of action nodes
-            nodes[i].value = values[i].GetValue()
+            if trigger_loop and hasattr(value, 'GetValue'):
+                # Info actions do not have GetValue
+                node.value = value.GetValue()
             # We store the 'RResultPtr's because,
             # those should be in scope while doing
             # a 'GetValue' call on them
-            nodes[i].ResultPtr = values[i]
+            node.ResultPtr = value
```

Comment:
Not all of these outputs are `RResultPtr`s. Of all the "other operations" of the ROOT RDataFrame class, only `Display` returns an `RResultPtr`.

Reply:
Furthermore, for those nodes that actually return `RResultPtr`s, we are storing both the pointer and the pointed result. Why do we need that?